packages/engine/core/src/lib/abstract.ts

```diff
@@ -77,48 +77,39 @@ export class Plugin<T extends Api = any, App extends ApiMap = any> implements Pl
     return this[method](...args)
   }
 
+  protected setCurrentRequest(request: PluginRequest) {
+    this.currentRequest = request
+  }
+
+  protected letContinue() {
+    delete this.currentRequest
+    this.queue = this.queue.filter((value) => {
```

Comment:
old requests aren't removed from the queue?

Reply:
they are